FHIR/FSHOnline
# Remove Bitly API Use

**Description:**

This PR removes our use of the Bitly API. In order to no longer tie our share functionality to an individual team member, we are going to remove the use of Bitly to shorten the link for sharing FSH. The functions that used the Bitly API are still in the code in case this functionality is wanted in the future. The functions should work (as long as the API still works the same) if anyone wants to use them in the future.

Now that there are no longer short links, the full-length long URL is now supported when loading in FSH snippets. The modal that opens when you go to share your FSH now scrolls properly so the box stays a reasonable size when the link is long. The "Copy" button now copies a formatted link, which is nice so that users aren't pasting really long URLs that look kind of ugly everywhere. If you copy the link, you should see it formatted properly as markdown in Zulip and as a hyperlink in places like Word.

In order to preserve existing short shared links, I grabbed all previously created links and added them as a file within `public/shares/`. Each file name is the short URL and the file content is the expanded URL. This should make it so that any share link that is out in the wild will still continue to work.

I removed the `VITE_BITLY_KEY` environment variable from the deploy script. I can also remove it from GitHub once this is ready to be merged.

**Testing Instructions:**

- Try creating a new share link and paste the formatted URL in Zulip and in other places, like Word.
  - I decided that if you didn't want a formatted version of the link, you could just copy the displayed link manually. However, if you think having an "unformatted copy" button would be helpful, let me know.
- Try loading a long link.
- Try loading on a previously shared (short) link.
- Take a look at the tests to make sure they cover the updated functionality.
- If you want, you can also build the app (`npm run build`) and try using the production build (`npm run preview`) to make sure the built app doesn't seem to slow down with all the new files in `public`.

There were lots of approaches that I tried for supporting existing short links, so if you don't like the approach with `public/shares/`, let me know and I'm happy to chat.

(Sorry that you have to scroll past 1645 files to get to the majority of the files that changed).

**Related Issue:**
N/A
